orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/ecs/EcsServerGroupCreator.groovy

```diff
@@ -57,6 +84,23 @@ class EcsServerGroupCreator implements ServerGroupCreator, DeploymentDetailsAwar
     if (operation.useTaskDefinitionArtifact) {
       if (operation.taskDefinitionArtifact) {
         operation.resolvedTaskDefinitionArtifact = getTaskDefArtifact(stage, operation.taskDefinitionArtifact)
+        if (operation.evaluateTaskDefinitionArtifactExpressions) {
+          Iterable<Object> rawArtifact =
+              retrySupport.retry(
+                  fetchAndParseArtifact(operation.resolvedTaskDefinitionArtifact), 10, Duration.ofMillis(200), true)
+
+          List<Map<Object, Object>> unevaluatedArtifact =
+              StreamSupport.stream(rawArtifact.spliterator(), false)
+                  .filter(Objects.&nonNull)
+                  .map(this.&coerceArtifactToList)
+                  .collect()
+                  .flatten()
+
+          Map<Object, Object> evaluatedArtifact = getSpelEvaluatedArtifact(unevaluatedArtifact, stage)
+
+          operation.spelProcessedTaskDefinitionArtifact = evaluatedArtifact
+          operation.evaluateTaskDefinitionArtifactExpressions = true
```

Comment:
why are we setting this to true? shouldn't it already be set to true in the customer's pipeline?

Reply:
extremely good catch 👍🏼. I will change this.